pkg/network/ebpf/c/tracer-events.h

```diff
@@ -14,7 +14,29 @@ static __always_inline int get_proto(conn_tuple_t *t) {
     return (t->metadata & CONN_TYPE_TCP) ? CONN_TYPE_TCP : CONN_TYPE_UDP;
 }
 
+static __always_inline void clean_protocol_classification(conn_tuple_t *tup) {
+    conn_tuple_t conn_tuple = *tup;
+    conn_tuple.pid = 0;
+    conn_tuple.netns = 0;
+    bpf_map_delete_elem(&connection_protocol, &conn_tuple);
+
+    conn_tuple_t *skb_tup_ptr = bpf_map_lookup_elem(&conn_tuple_to_socket_skb_conn_tuple, &conn_tuple);
+    if (skb_tup_ptr != NULL) {
+        conn_tuple_t skb_tup = *skb_tup_ptr;
```

Comment:
Do we need `skb_tup`?

Reply:
yes, we save the protocol for both `sock_conn_tuple` and `skb_conn_tuple` (see the documentation in `protocol-classification-maps` above the maps `skb_conn_tuple_to_socket_conn_tuple` and `conn_tuple_to_socket_skb_conn_tuple`)

should I write that explanation in the cleaning function as well?

Reply:
Sorry I meant its not being used anywhere other than on line 31; could you just use `skb_tup_ptr` there?

Reply:
oh, no
some old verifiers rejects it as `skb_tup_ptr` is map_value while it expected to have `fp`, etc.